ewatercycle/models/lisflood.py

```diff
@@ -241,6 +241,53 @@ def get_value_as_xarray(self, name: str) -> xr.DataArray:
 
         return da
 
+    def _coords_to_indices(self, name: str, lat: Iterable[float], lon: Iterable[float]) -> Iterable[int]:
+        """Convert lat, lon coordinates into model indices."""
+        grid_id = self.bmi.get_var_grid(name)
+        shape = self.bmi.get_grid_shape(grid_id) # shape returns (len(y), len(x))
+        x_model = self.bmi.get_grid_x(grid_id)
+        y_model = self.bmi.get_grid_y(grid_id)
+        spacing_model = self.bmi.get_grid_spacing(grid_id)
+
+        # Create a grid from coordinates
+        x_vectors, y_vectors = np.meshgrid(x_model, y_model)
+
+        indices = []
+        # in lisflood, x corresponds to lon, and y to lat.
+        # this might not be the case for other models!
+        for x, y in zip(lon, lat):
+            # here we use Euclidean distance, but it is not accurate as the coordinates are in lon/lat and degrees.
+            distance = ((x_vectors - x) ** 2 + (y_vectors - y) ** 2) ** (1/2)
+            index = distance.argmin()
+            indices.append(index)
+            idy, idx = np.unravel_index(index, shape)
+            coord_converted = (round(x_model[idx], 4), round(y_model[idy], 4)) # use 4 digits in round
+            coord_user = (round(x, 4), round(y, 4))
+            # Provide feedback
+            print(f"Your coordinates {coord_user} are closest to these model coordinates {coord_converted}.")
```

Comment:
I think @sverhoeven preferred not to use print statements in our source code. An alternative could be to return the converted coords as a second return value.

Reply:
right, it is refactored.